plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/AlwaysExternalCustomServerEvaluationStrategy.java

```diff
@@ -49,11 +53,26 @@ public AlwaysExternalCustomServerEvaluationStrategy(
         chePort,
         true,
         cheWorkspacesRoutingSuffixProvider);
+    this.cheDockerCustomExternalProtocol = cheDockerCustomExternalProtocol;
   }
 
   @Override
   protected Map<String, String> getInternalAddressesAndPorts(
       ContainerInfo containerInfo, String internalHost) {
     return super.getExternalAddressesAndPorts(containerInfo, internalHost);
   }
+
+  @Override
+  protected ServerImpl updateServer(ServerImpl server) {
+    super.updateServer(server);
+    if (!Strings.isNullOrEmpty(cheDockerCustomExternalProtocol)) {
+    ServerPropertiesImpl properties = (ServerPropertiesImpl) server.getProperties();
+    String url = properties.getInternalUrl();
+    if(url != null && url.startsWith("http:")) {
```

Comment:
@davidfestal formatting in `if(`.  Probably it would be better to extract it to some method e.g. `isHttp(String url)`
Could you please clarify what exactly is done in `updateServer` and why internal URL is updated only when it is http based ?